services/audit-service/src/main/java/datawave/microservice/audit/health/rabbit/RabbitHealthChecker.java

```diff
@@ -0,0 +1,705 @@
+    private static Queue createQueue(QueueProperties qp) {
+        return new Queue(qp.getName(), qp.isDurable(), qp.isExclusive(), qp.isAutoDelete(), qp.getArguments());
+    }
+    
+    /**
+     * Creates an {@link Binding} from the configured {@link BindingProperties}
+     * 
+     * @param bp
+     *            The binding properties derived from the {@link RabbitHealthProperties}
+     * @return A {@link Binding}
+     */
+    private static Binding createBinding(BindingProperties bp) {
+        Binding.DestinationType destType;
+        switch (bp.getDestinationType().toLowerCase()) {
+            case "queue":
+                destType = Binding.DestinationType.QUEUE;
+                break;
+            case "exchange":
+                destType = Binding.DestinationType.EXCHANGE;
+                break;
+            default:
+                throw new RuntimeException("Unable to create binding given properties: [" + bp + "]");
+        }
+        return new Binding(bp.getDestination(), destType, bp.getExchange(), bp.getRoutingKey(), bp.getArguments());
+    }
+    
+    /**
+     * Determines which poll interval should be used, based on the overall RabbitMQ health
+     * 
+     * @return the poll interval to use in millis
+     */
+    @Override
+    public long pollIntervalMillis() {
+        if (isHealthy())
+            return rabbitHealthProperties.getHealthyPollIntervalMillis();
+        else
+            return rabbitHealthProperties.getUnhealthyPollIntervalMillis();
+    }
+    
+    /**
+     * Performs a health check of RabbitMQ.
+     * <p>
+     * This method is synchronized with the isHealthy method to ensure that the health check is given time to complete before deciding whether to accept an
+     * audit request via the {@link AuditController}
+     * <p>
+     * RabbitMQ is considered to be healthy if the cluster, exchanges, queues, and bindings are configured as specified in the {@link RabbitHealthProperties}.
+     * If we are not able to determine the health of the cluster, exchanges, queues, and bindings (perhaps due to the management API being unavailable, or the
+     * cluster being down), we will consider RabbitMQ to be unhealthy.
+     */
+    @Override
+    public void runHealthCheck() {
+        initRabbitClient();
+        
+        synchronized (this) {
+            log.trace("RabbitMQ Health Check - Started");
+            
+            boolean wasHealthy = isHealthy();
+            
+            // check the cluster
+            clusterHealthCheck();
+            
+            // check the exchanges
+            exchangesHealthCheck();
+            
+            // check the queues
+            queuesHealthCheck();
+            
+            // check the bindings
+            bindingsHealthCheck();
+            
+            boolean isHealthy = isHealthy();
+            
+            if (wasHealthy) {
+                if (isHealthy) {
+                    log.info("RabbitMQ is still healthy.");
```

Comment:
Health check might run very frequently. Do we want to log every successful health check to info?

Reply:
I was conflicted on this.  Looking at it again, maybe try this...  Use INFO for transitions from unhealthy to healthy.  Use WARN for transitions from healthy to unhealthy.  Everything else is trace.  Sound good?

Reply:
Yup, that works. I'd say leave this log, but move it to debug level.